cpp/include/ray/api/function_manager.h

```diff
@@ -223,19 +231,13 @@ class FunctionManager {
     return instance;
```

Comment:
It is better to never destruct instance, e.g. `static auto* instance = new FunctionManager()`. Otherwise `instance` destructor would run during program shutdown and maybe library unloading, which may cause hard to debug errors.

Reply:
Good point! But do you mean make `FunctionHelper` never destructed? We save `boost::dll::shared_library` in `FunctionHelper`, not `FunctionManager `.

Reply:
Not destroying the container of `boost::dll::shared_library` is a great point. Anytime we need to have a static function local and it is not trivially destructible, it is recommended to store a pointer so it is never destructed. The argument is that otherwise if there is another thread reading the variable, the read can race with the static local variable's destruction.